server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceAggregationBuilder.java

```diff
@@ -174,8 +147,10 @@ protected final void doWriteTo(StreamOutput out) throws IOException {
     protected abstract void innerWriteTo(StreamOutput out) throws IOException;
 
     /**
-     * Should this builder serialize its targetValueType? Defaults to false. All subclasses that override this to true should use the three
-     * argument read constructor rather than the four argument version.
+     * DO NOT OVERRIDE THIS!
+     *
+     * This method only exists for legacy support.  No new aggregations need this, nor should they override it.
+     *
```

Comment:
Maybe add the `@deprecated` annotation too?

Reply:
Actually, do any aggs override this today? Can we just nuke it?

Reply:
Yes, several aggs do override this, but even before this PR, they were just serializing nulls.

Reply:
Gotcha